mantidproject/mantid
# Reflectometry interface: Introduce autoreduction button

As described in the attached issue, this PR adds a button to perform autoreduction. When supplied an investigation id and the button is clicked, the following occurs:

- Obtain runs that are part of the investigation the id was supplied for.
- Transfer all runs to the processing table.
- Select all runs and process them.

When autoreduction is running, the button is disabled. Pausing reduction will re-enable the button. If the instrument, investigation id or transfer strategy is changed while paused before clicking 'autoreduce' again, a new autoreduction (including searching for runs and transfer) will be started. Otherwise it simply resumes reducing runs just like `Process` would.

~_Wait for PR #19977 to be merged first before merging this._~

**To test:**
- Code review.
- Ensure documentation is updated and sensible.
- Manual testing steps:
  - Login into the ICAT catalog.
  - The interface can be accessed from Interfaces->Reflectometry->ISIS Reflectometry.
  - Select `POLREF`, enter investigation id `1520249` and click on the `autoreduce` button to begin autoreduction.
  - Ensure that changing id, instrument or transfer strategy causes a new autoreduction to be started.
  - Currently it is possible to cause critical errors by clicking on buttons such as `delete row` while reduction is in progress. These will be dealt with in #19951.

Fixes #19453.

**Release Notes** 

*Release notes have been updated.*

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####


##### Functional Tests #####


Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
